pkg/cgroups/cgroups_linux.go

```diff
@@ -45,6 +49,51 @@ func mountCgroup() error {
 	return nil
 }
 
+// cgrpSanityCheck performs a sanity check on the mounted cgroup fs and issues
+// a warning if something seems wrong.
+//
+// It does this by recursively checking the number of pids in "cgroup.procs". If they are 2
+// or less, it is likely that the cilium pod is the only container in this
+// cgroup.
+func cgrpSanityCheck() {
+
+	procCount := 0
+	err := filepath.Walk(cgroupRoot, func(path string, info fs.FileInfo, err error) error {
+		if err != nil {
+			log.WithField("path", path).WithError(err).Info("cgrpSanityCheck: error traversing cgroup directory")
+		}
+
+		if info.IsDir() {
+			return nil
+		}
+
+		if info.Name() != "cgroup.procs" {
+			return nil
+		}
+
+		f, err := os.Open(path)
+		if err != nil {
+			log.WithField("path", path).WithError(err).Info("cgrpSanityCheck: failed to open file")
+			return nil
+		}
+		s := bufio.NewScanner(f)
+		for s.Scan() {
+			procCount++
+			if procCount > 2 {
+				f.Close()
+				return errors.New("")
+			}
+		}
+		f.Close()
+		return nil
+	})
+
+	// if error is not set, it means that we did not reached the desired procCount
+	if err == nil {
+		log.Warnf("%s has only %d process(es) under it (see: https://docs.cilium.io/en/latest/warnings.html#cgrouplowproccount)", cgroupRoot, procCount)
```

Comment:
If we intend to reuse this mechanism to point to the documentation for warnings, I would like to avoid hardcoding the URL everytime. Ideally, there should be some helper function to generate the link from the warning ID, and that we could call from elsewhere in the code.

Reply:
Makes sense. It would also be the proper place to deal any versioning detection we want to do.